CRM/Cmsuser/Form/Setting.php

```diff
@@ -23,7 +23,12 @@ public function buildQuickForm() {
     $this->add('advcheckbox', 'cmsuser_login_immediately', ts('Login User Immediately?'));
     $this->add('advcheckbox', 'cmsuser_allow_existing_user_login', ts('Allow existing User to auto login?'));
     if (CIVICRM_UF == 'Drupal8') {
-      $user_role_names = user_role_names(TRUE);
+    // Borrowed from https://www.drupal.org/files/issues/2024-01-05/3412456.patch
+    $user_roles = Role::loadMultiple();
+    $user_role_names = [];
+      foreach ($user_roles as $role) {
+       $user_role_names[$role->id()] = $role->label();
+      }      
       unset($user_role_names['authenticated']);
```

Comment:
Shouldn't the "anonymous" be unset here - makes more sense at new registered user account creation?